jerry-core/ecma/builtin-objects/ecma-builtins.c

```diff
@@ -51,6 +51,74 @@ typedef ecma_value_t (*ecma_builtin_dispatch_routine_t)(uint16_t builtin_routine
  */
 typedef ecma_value_t (*ecma_builtin_dispatch_call_t)(const ecma_value_t arguments_list[],
                                                      ecma_length_t arguments_number);
+/**
+ * Definition of a builtin descriptor which contains the builtin object's:
+ * - prototype objects's id (8-bits)
+ * - type (3-bits)
+ * - [[Extensible]] property's value (1-bit)
+ *
+ * Layout:
+ *
+ * |---------------------|-------------|---------------|--------------------|
+ *     prototype_id(8)      unused(4)     obj_type(3)     is_extensible(1)
```

Comment:
I would put the obj_type to last, and the flag before it. Are you sure 8 bit for prototype id is enough? I would use the 4 unused bit for it as well, and if we need more bits we can figure something. Btw is there an object which is not extensible by default?

Reply:
OK, let's swap them.

I think 8 bit is more than enough for the prototype id. Currently we have 60 builtin objects (with es2015 profile) and only 7 of these used as a prototype.

Yes, the [ [[ThrowTypeError]] ](https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/ecma/builtin-objects/ecma-builtins.inc.h#L265) object is not extensible.